### PR TITLE
[BugFix](DOE) Add http prefix when it's not set in hosts properties.

### LINKED ISSUE #17722: [Enhancement] (DOE) hosts protocol access exception
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

1、create catalog
 CREATE CATALOG es_test PROPERTIES (
    "type"="es",
    "hosts"="127.0.0.1:9200"
);
The table information can be queried.

2、Query error
ERROR 1105 (HY000): errCode = 2, detailMessage = Failed to connect to ES server, errmsg is: URL using bad/illegal format or missing URL
image

Change hosts to "http://127.0.0.1:9200", it works.

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SaintBacchus**
# Proposed changes

Issue Number: close #17722

## Problem summary

Add http prefix when it's not set in hosts properties.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by SaintBacchus** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.82 seconds
 stream load tsv:          448 seconds loaded 74807831229 Bytes, about 159 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230313101833_clickbench_pr_113524.html

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/catalog/EsTable.java [hunk 1]
```diff
@@ -153,6 +153,12 @@ private void validate(Map<String, String> properties) throws DdlException {
         EsResource.valid(properties, false);
         hosts = properties.get(EsResource.HOSTS).trim();
         seeds = hosts.split(",");
+        for (int i = 0; i < seeds.length; i++) {
+            String seed = seeds[i].trim();
+            if (!seed.startsWith("http://") && !seed.startsWith("https://")) {
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/EsTable.java:158]: I think need judge whether config ssl.

**Comment by SaintBacchus** [comment]
run buildall

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalCatalog.java [hunk 2]
```diff
@@ -70,7 +70,10 @@ private Map<String, String> processCompatibleProperties(Map<String, String> prop
 
     public String[] getNodes() {
         String hosts = catalogProperty.getOrDefault(EsResource.HOSTS, "");
-        return hosts.trim().split(",");
+        String sslEnabled = catalogProperty.getOrDefault(EsResource.HTTP_SSL_ENABLED, "false");
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalCatalog.java:73]: "false" can use HTTP_SSL_ENABLED_DEFAULT_VALUE

**Review by stalary — DISMISSED** [review]
LGTM

**Comment by stalary** [comment]
run buildall

**Comment by SaintBacchus** [comment]
run p1

**Comment by SaintBacchus** [comment]
run p1

**Comment by stalary** [comment]
run p0

**Comment by SaintBacchus** [comment]
run p1

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run p1

**Comment by SaintBacchus** [comment]
run buildall

**Comment by SaintBacchus** [comment]
run clickbench

**Comment by SaintBacchus** [comment]
run clickbench

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run clickbench

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run clickbench

**Comment by SaintBacchus** [comment]
run clickbench

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run buildall

**Comment by SaintBacchus** [comment]
run buildall

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run arm

**Comment by SaintBacchus** [comment]
run buildall

**Comment by SaintBacchus** [comment]
run buildall

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run p0

**Comment by SaintBacchus** [comment]
run arm

**Review by stalary — APPROVED** [review]
LGTM

**Pull request merged**